Crash on MC 1.18.1

Fabric 0.12.12
Fabric API: 0.45.0
Architectury API: 3.3.2
REI: 7.1.357

crash only happens when adding architectury api back to mods folder.
crash report:
```
---- Minecraft Crash Report ----
// I bet Cylons wouldn't have this problem.

Time: 12/26/21, 8:06 PM
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'fabric-networking-v0'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:452)
	at net.minecraft.client.main.Main.main(Main.java:199)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	Suppressed: java.lang.RuntimeException: Failed to initialize REI entry point: me.shedaniel.rei.RoughlyEnoughItemsCore
		at me.shedaniel.rei.RoughlyEnoughItemsInitializer.initializeEntryPoint(RoughlyEnoughItemsInitializer.java:80)
		at me.shedaniel.rei.RoughlyEnoughItemsInitializer.onInitialize(RoughlyEnoughItemsInitializer.java:40)
		at java.base/java.lang.invoke.MethodHandleProxies$1.invoke(MethodHandleProxies.java:198)
		at jdk.proxy2/com.sun.proxy.jdk.proxy2.$Proxy31.onInitialize(Unknown Source)
		at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
		... 11 more
	Caused by: java.lang.BootstrapMethodError: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3244 failed
		at net.fabricmc.fabric.impl.networking.ServerSidePacketRegistryImpl.register(ServerSidePacketRegistryImpl.java:68)
		at dev.architectury.networking.fabric.NetworkManagerImpl.registerC2SReceiver(NetworkManagerImpl.java:60)
		at dev.architectury.networking.fabric.NetworkManagerImpl.registerReceiver(NetworkManagerImpl.java:51)
		at dev.architectury.networking.NetworkManager.registerReceiver(NetworkManager.java)
		at me.shedaniel.rei.RoughlyEnoughItemsNetwork.onInitialize(RoughlyEnoughItemsNetwork.java:56)
		at me.shedaniel.rei.RoughlyEnoughItemsCore.onInitialize(RoughlyEnoughItemsCore.java:154)
		at me.shedaniel.rei.RoughlyEnoughItemsInitializer.initializeEntryPoint(RoughlyEnoughItemsInitializer.java:76)
		... 15 more
	Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3244 failed
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:252)
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150)
		at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
		... 22 more
	Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
		at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
		at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
		at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:247)
		... 25 more
	Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: LVT in net/minecraft/class_3244::method_31286(Lnet/minecraft/class_5513$class_5837;)V has incompatible changes at opcode 115 in callback architectury.mixins.json:MixinServerGamePacketListenerImpl->@Inject::handleChat(Lnet/minecraft/class_5513$class_5837;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;Ljava/lang/String;Lnet/minecraft/class_2561;Lnet/minecraft/class_2561;)V from mod architectury.
 Expected: [Ljava/lang/String;, Lnet/minecraft/class_2561;, Lnet/minecraft/class_2561;]
    Found: [Ljava/lang/String;, Ljava/lang/String;, Lnet/minecraft/class_2561;]
Available: [Ljava/lang/String;, Ljava/lang/String;, Lnet/minecraft/class_2561;, Lnet/minecraft/class_2561;]
		at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:545)
		at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:495)
		at org.spongepowered.asm.mixin.injection.code.Injector.inject(Injector.java:276)
		at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.inject(InjectionInfo.java:445)
		at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1374)
		at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052)
		at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400)
		at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325)
		at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421)
		at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
		at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
		... 28 more
Caused by: java.lang.BootstrapMethodError: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3244 failed
	at net.fabricmc.fabric.api.networking.v1.S2CPlayChannelEvents.lambda$static$1(S2CPlayChannelEvents.java:36)
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.update(ArrayBackedEvent.java:57)
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.<init>(ArrayBackedEvent.java:53)
	at net.fabricmc.fabric.impl.base.event.EventFactoryImpl.createArrayBacked(EventFactoryImpl.java:44)
	at net.fabricmc.fabric.api.event.EventFactory.createArrayBacked(EventFactory.java:63)
	at net.fabricmc.fabric.api.networking.v1.S2CPlayChannelEvents.<clinit>(S2CPlayChannelEvents.java:36)
	at net.fabricmc.fabric.impl.networking.OldNetworkingHooks.onInitialize(OldNetworkingHooks.java:27)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 11 more
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3244 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:252)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 19 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:247)
	... 22 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: LVT in net/minecraft/class_3244::method_31286(Lnet/minecraft/class_5513$class_5837;)V has incompatible changes at opcode 115 in callback architectury.mixins.json:MixinServerGamePacketListenerImpl->@Inject::handleChat(Lnet/minecraft/class_5513$class_5837;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;Ljava/lang/String;Lnet/minecraft/class_2561;Lnet/minecraft/class_2561;)V from mod architectury.
 Expected: [Ljava/lang/String;, Lnet/minecraft/class_2561;, Lnet/minecraft/class_2561;]
    Found: [Ljava/lang/String;, Ljava/lang/String;, Lnet/minecraft/class_2561;]
Available: [Ljava/lang/String;, Ljava/lang/String;, Lnet/minecraft/class_2561;, Lnet/minecraft/class_2561;]
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:545)
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:495)
	at org.spongepowered.asm.mixin.injection.code.Injector.inject(Injector.java:276)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.inject(InjectionInfo.java:445)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1374)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 25 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:452)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Advanced Windows 32 Base API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:User Experience Controls Library:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Crypto API32:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.19041.867 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS Client API DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.19041.1202 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP Helper API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Windows NT BASE API Client DLL:10.0.19041.1348 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Windows NT BASE API Client DLL:10.0.19041.1348 (WinBuild.160101.0800):Microsoft Corporation
		MpOav.dll:IOfficeAntiVirus Module:4.18.2111.5 (WinBuild.160101.0800):Microsoft Corporation
		NLAapi.dll:Network Location Awareness 2:10.0.19041.1151 (WinBuild.160101.0800):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.19041.610 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE for Windows:10.0.19041.1320 (WinBuild.160101.0800):Microsoft Corporation
		OleAut32.dll:OLEAUT32.DLL:10.0.19041.985 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Performance Data Helper DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remote Procedure Call Runtime:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Windows Shell Common Dll:10.0.19041.964 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Multi-User Windows USER API Client DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.29.30133.0 built by: vcwrkspc:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP Services:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0 32-Bit DLL:10.0.19041.1081 (WinBuild.160101.0800):Microsoft Corporation
		WSOCK32.dll:Windows Socket 32-Bit DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		Wldp.dll:Windows Lockdown Policy:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.19041.746 (WinBuild.160101.0800):Microsoft Corporation
		bcrypt.dll:Windows Cryptographic Primitives Library:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.19041.1415 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM for Windows:10.0.19041.1320 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc.DLL:DHCP Client Service:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc6.DLL:DHCPv6 Client:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		fwpuclnt.dll:FWP/IPsec User-Mode API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.19041.1387 (WinBuild.160101.0800):Microsoft Corporation
		glfw.dll
		java.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		javaw.exe:OpenJDK Platform binary:17.0.1.0:Microsoft
		jemalloc.dll
		jimage.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		jli.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		jna15340743048423213717.dll:JNA native library:6.1.1:Java(TM) Native Access (JNA)
		jvm.dll:OpenJDK 64-Bit server VM:17.0.1.0:Microsoft
		kernel.appcore.dll:AppModel API Host:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		management.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		management_ext.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		msvcp140.dll:Microsoft® C Runtime Library:14.29.30133.0 built by: vcwrkspc:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Microsoft Windows Sockets 2.0 Service Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:E-mail Naming Shim Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Windows NCrypt Router:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		nio.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		ntdll.dll:NT Layer DLL:10.0.19041.1023 (WinBuild.160101.0800):Microsoft Corporation
		perfos.dll:Windows System Performance Objects DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		pnrpnsp.dll:PNRP Name Space Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.19041.844 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Shell Light-weight Utility Library:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		svml.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.29.30133.0 built by: vcwrkspc:Microsoft Corporation
		verify.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
		win32u.dll:Win32u:10.0.19041.1387 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:Microsoft WinRT Storage API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:OpenJDK Platform binary:17.0.1.0:Microsoft
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:199)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 1.18.1
	Minecraft Version ID: 1.18.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 1561092568 bytes (1488 MiB) / 2147483648 bytes (2048 MiB) up to 4294967296 bytes (4096 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
	Identifier: Intel64 Family 6 Model 158 Stepping 10
	Microarchitecture: Coffee Lake
	Frequency (GHz): 3.19
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: NVIDIA GeForce GTX 1050 Ti
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x1c82
	Graphics card #0 versionInfo: DriverVersion=27.21.14.5751
	Graphics card #1 name: Intel(R) UHD Graphics 630
	Graphics card #1 vendor: Intel Corporation (0x8086)
	Graphics card #1 VRAM (MB): 1024.00
	Graphics card #1 deviceId: 0x3e92
	Graphics card #1 versionInfo: DriverVersion=27.20.100.9664
	Memory slot #0 capacity (MB): 16384.00
	Memory slot #0 clockSpeed (GHz): 2.67
	Memory slot #0 type: DDR4
	Virtual memory max (MB): 22850.69
	Virtual memory used (MB): 12535.04
	Swap memory total (MB): 6656.00
	Swap memory used (MB): 143.74
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx4G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		architectury: Architectury 3.3.2
		blur: Blur (Fabric) 2.4.1
		cardinal-components-base: Cardinal Components API (base) 4.0.0
		cardinal-components-entity: Cardinal Components API (entities) 4.0.0
		cloth-basic-math: cloth-basic-math 0.6.0
		cloth-config: Cloth Config v6 6.1.48
		collective-fabric: Collective (Fabric) 3.14
		despawningeggshatch-fabric: Despawning Eggs Hatch (Fabric) 2.3
		difficultylock-fabric: Difficulty Lock (Fabric) 1.3
		elytratrinket: Elytra Trinket 3.0.0-1.18.1
		fabric: Fabric API 0.45.0+1.18
		fabric-api-base: Fabric API Base 0.4.1+b4f4f6cd65
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.5.0+17be577f65
		fabric-biome-api-v1: Fabric Biome API (v1) 6.0.1+ded849a965
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.9+3ac43d9565
		fabric-command-api-v1: Fabric Command API (v1) 1.1.6+3ac43d9565
		fabric-commands-v0: Fabric Commands (v0) 0.2.5+b4f4f6cd65
		fabric-containers-v0: Fabric Containers (v0) 0.1.18+d154e2c665
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.4.5+6f53a73d65
		fabric-copper-golem: Copper Golem 1.0.3+1.18
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.8+3ac43d9565
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.7+43d2957165
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.5+6b21378a65
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.16+bfa23f1765
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.6+b4f4f6cd65
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.10+3ac43d9565
		fabric-item-api-v1: Fabric Item API (v1) 1.3.0+691a79b565
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.3+3ac43d9565
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.8+c8aba2f365
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.6+b4f4f6cd65
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.10+c15ca33565
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.8+3ac43d9565
		fabric-mining-level-api-v1: Fabric Mining Level API (v1) 1.0.3+3ac43d9565
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.7+b4f4f6cd65
		fabric-models-v0: Fabric Models (v0) 0.3.3+3ac43d9565
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.18+3ac43d9565
		fabric-networking-v0: Fabric Networking (v0) 0.3.5+b4f4f6cd65
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.11.0+3b82842e65
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.8+3ac43d9565
		fabric-particles-v1: Fabric Particles (v1) 0.2.9+526dc1ac65
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.8.6+533be9ba65
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.9+3ac43d9565
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.12+3ac43d9565
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.7+b4f4f6cd65
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.4+7242e9d765
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.18+3ac43d9565
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.9+b4f4f6cd65
		fabric-rendering-v1: Fabric Rendering (v1) 1.10.3+6b21378a65
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.11+3ac43d9565
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.7+3ac43d9565
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.11+3ac43d9565
		fabric-structure-api-v1: Fabric Structure API (v1) 2.0.8+295197a765
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.2.5+3ac43d9565
		fabric-textures-v0: Fabric Textures (v0) 1.0.9+3ac43d9565
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.3.4+7de09f5565
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.5.6+b4f4f6cd65
		fabricloader: Fabric Loader 0.12.12
		fiber: fiber 0.23.0-2
		java: OpenJDK 64-Bit Server VM 17
		midnightlib: MidnightLib 0.3.1
		minecraft: Minecraft 1.18.1
		pling: Pling 1.5.0
		roughlyenoughitems: Roughly Enough Items 7.1.357
		satin: Satin 1.7.0
		sit: Sit 1.18.1-15
		trinketofundying: Trinket of Undying 2.0.1-1.18
		trinkets: Trinkets 3.1.0
		voidtotem: VoidTotem (Fabric) 1.18.1-1.1.0
	Launched Version: fabric-loader-0.12.12-1.18.1
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Unknown
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>
```

--- a/fabric/src/main/java/dev/architectury/mixin/fabric/MixinChunkSerializer.java
+++ b/fabric/src/main/java/dev/architectury/mixin/fabric/MixinChunkSerializer.java
@@ -19,6 +19,7 @@
 
 package dev.architectury.mixin.fabric;
 
+import com.mojang.serialization.Codec;
 import dev.architectury.event.events.common.ChunkEvent;
 import net.minecraft.core.Registry;
 import net.minecraft.nbt.CompoundTag;
@@ -26,7 +27,14 @@ import net.minecraft.nbt.ListTag;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.ai.village.poi.PoiManager;
 import net.minecraft.world.level.ChunkPos;
-import net.minecraft.world.level.chunk.*;
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.chunk.ChunkAccess;
+import net.minecraft.world.level.chunk.ChunkSource;
+import net.minecraft.world.level.chunk.ChunkStatus;
+import net.minecraft.world.level.chunk.LevelChunkSection;
+import net.minecraft.world.level.chunk.PalettedContainer;
+import net.minecraft.world.level.chunk.ProtoChunk;
+import net.minecraft.world.level.chunk.UpgradeData;
 import net.minecraft.world.level.chunk.storage.ChunkSerializer;
 import net.minecraft.world.level.levelgen.blending.BlendingData;
 import net.minecraft.world.level.lighting.LevelLightEngine;
@@ -40,9 +48,9 @@ import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
 public class MixinChunkSerializer {
     @Inject(method = "read", at = @At("RETURN"), locals = LocalCapture.CAPTURE_FAILHARD)
     private static void load(ServerLevel serverLevel, PoiManager poiManager, ChunkPos chunkPos, CompoundTag compoundTag,
-                             CallbackInfoReturnable<ProtoChunk> cir, UpgradeData upgradeData,
+                             CallbackInfoReturnable<ProtoChunk> cir, ChunkPos chunkPos2, UpgradeData upgradeData,
                              boolean bl, ListTag listTag, int i, LevelChunkSection levelChunkSections[], boolean bl2, ChunkSource chunkSource,
-                             LevelLightEngine levelLightEngine, Registry registry, long m, ChunkStatus.ChunkType chunkType,
+                             LevelLightEngine levelLightEngine, Registry registry, Codec<PalettedContainer<Biome>> codec, long m, ChunkStatus.ChunkType chunkType,
                              BlendingData blendingData, ChunkAccess chunkAccess2) {
         ChunkEvent.LOAD_DATA.invoker().load(chunkAccess2, serverLevel, compoundTag);
     }

--- a/fabric/src/main/java/dev/architectury/mixin/fabric/MixinPatrolSpawner.java
+++ b/fabric/src/main/java/dev/architectury/mixin/fabric/MixinPatrolSpawner.java
@@ -24,6 +24,7 @@ import net.minecraft.core.BlockPos;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.MobSpawnType;
 import net.minecraft.world.entity.monster.PatrollingMonster;
+import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.levelgen.PatrolSpawner;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
@@ -48,7 +49,7 @@ public abstract class MixinPatrolSpawner {
             cancellable = true,
             locals = LocalCapture.CAPTURE_FAILHARD
     )
-    private void checkPatrolSpawn(ServerLevel level, BlockPos pos, Random r, boolean b, CallbackInfoReturnable<Boolean> cir, PatrollingMonster entity) {
+    private void checkPatrolSpawn(ServerLevel level, BlockPos pos, Random r, boolean b, CallbackInfoReturnable<Boolean> cir, BlockState blockState, PatrollingMonster entity) {
         var result = EntityEvent.LIVING_CHECK_SPAWN.invoker().canSpawn(entity, level, pos.getX(), pos.getY(), pos.getZ(), MobSpawnType.PATROL, null);
         if (result.value() != null) {
             cir.setReturnValue(result.value());

--- a/fabric/src/main/java/dev/architectury/mixin/fabric/MixinPhantomSpawner.java
+++ b/fabric/src/main/java/dev/architectury/mixin/fabric/MixinPhantomSpawner.java
@@ -23,12 +23,15 @@ import dev.architectury.event.events.common.EntityEvent;
 import net.minecraft.core.BlockPos;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.stats.ServerStatsCounter;
 import net.minecraft.world.DifficultyInstance;
 import net.minecraft.world.entity.MobSpawnType;
 import net.minecraft.world.entity.SpawnGroupData;
 import net.minecraft.world.entity.monster.Phantom;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.levelgen.PhantomSpawner;
+import net.minecraft.world.level.material.FluidState;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
@@ -53,8 +56,9 @@ public abstract class MixinPhantomSpawner {
             locals = LocalCapture.CAPTURE_FAILSOFT // SOFT, because this will break in 2 seconds
     )
     private void checkPhantomSpawn(ServerLevel level, boolean bl, boolean bl2, CallbackInfoReturnable<Integer> cir,
-                                   Random random, int i, Iterator<ServerPlayer> it, Player player, BlockPos pos, DifficultyInstance diff, BlockPos pos2,
-                                   SpawnGroupData sgd, int l, int m, Phantom entity) {
+                                   Random random, int i, Iterator<ServerPlayer> it, Player player, BlockPos pos, DifficultyInstance diff,
+                                   ServerStatsCounter serverStatsCounter, int j, int k, BlockPos pos2,
+                                   BlockState blockState, FluidState fluidState, SpawnGroupData sgd, int l, int m, Phantom entity) {
         if (EntityEvent.LIVING_CHECK_SPAWN.invoker().canSpawn(entity, level, pos.getX(), pos.getY(), pos.getZ(), MobSpawnType.NATURAL, null).value() == Boolean.FALSE) {
             cir.setReturnValue(0);
             cir.cancel();

--- a/fabric/src/main/java/dev/architectury/mixin/fabric/MixinServerGamePacketListenerImpl.java
+++ b/fabric/src/main/java/dev/architectury/mixin/fabric/MixinServerGamePacketListenerImpl.java
@@ -57,7 +57,7 @@ public abstract class MixinServerGamePacketListenerImpl {
             at = @At(value = "INVOKE",
                     target = "Lnet/minecraft/server/players/PlayerList;broadcastMessage(Lnet/minecraft/network/chat/Component;Ljava/util/function/Function;Lnet/minecraft/network/chat/ChatType;Ljava/util/UUID;)V"),
             cancellable = true, locals = LocalCapture.CAPTURE_FAILHARD)
-    private void handleChat(TextFilter.FilteredText message, CallbackInfo ci, String string, Component component, Component component2) {
+    private void handleChat(TextFilter.FilteredText message, CallbackInfo ci, String string, String string2, Component component, Component component2) {
         var chatComponent = new ChatComponentImpl(component2, component);
         var process = ChatEvent.SERVER.invoker().process(this.player, message, chatComponent);
         if (process.isEmpty()) return;

--- a/fabric/src/main/java/dev/architectury/mixin/fabric/client/MixinGameRenderer.java
+++ b/fabric/src/main/java/dev/architectury/mixin/fabric/client/MixinGameRenderer.java
@@ -39,7 +39,7 @@ import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
 import java.util.List;
 import java.util.function.Consumer;
 
-@Mixin(GameRenderer.class)
+@Mixin(value = GameRenderer.class, priority = 1100)
 public abstract class MixinGameRenderer {
     @Shadow
     @Final

